Adding DM7 signal to NEXAFS data collection

PriyankaKetkarBNL opened the pull request:
I created a DM7_Photodiode_int signal in the signals.py file and added this signal to the list of signals/detectors in the energyscancore.py file.  

I mainly looked for places where Beamstop_WAXS_int was used, and added the DM7_Photodiode_int signal to the list.  I also saw  Beamstop_WAXS_int in fly_alignment.py (fly_find_fiducials function), alignment.py (spiral scans), and common_procedures.py (buildeputable), but I did not add the DM7_Photodiode_int signal to these files, as it did not seem relevant.  I also did not add the DM7_Photodiode_int signal into lines 626, 627, 636, and 637 of energyscancore.py for the same reason.  Let me know if I need to adjust anything.

EliotGann commented on rsoxs/Functions/energyscancore.py:
```diff
@@ -366,7 +366,7 @@ def new_en_scan_core(
     sim_mode=False,  # if true, check all inputs but do not actually run anything
     md=None,  # md to pass to the scan
-    signals = [Beamstop_WAXS_int, Beamstop_SAXS_int, Izero_Mesh_int, Sample_TEY_int,mono_en,epu_gap,ring_current],
+    signals = [Beamstop_WAXS_int, Beamstop_SAXS_int, DM7_Photodiode_int, Izero_Mesh_int, Sample_TEY_int,mono_en,epu_gap,ring_current],
```
I don't think it's relevant to measure in an RSoXS scan?  generally the detector being in makes this not really useful, right?

PriyankaKetkarBNL replied:
Oh, I see, I took it back out.  Maybe not in this pull request, but do you think it makes sense down the line to rename the function so it is more obvious that it is an RSoXS scan?  The function name gave me the impression that it was just a general energy scan universally applicable to RSoXS or NEXAFS.

EliotGann commented:
1. I think a more relevant user name might be useful, as no one (even me a week from now) will remember what DM7 is.  I might suggest changing the "name" field to be something more relevant and universally recognizable.
2. This is indeed probably very relevant to those other positions where WAXS_beamstop is being used now.  in particular for alignment of fiducials (if it is a more reliable transmission signal) and building EPU tables.
3. A possible consequence of adding another signal might be a slow down in acquisitions.  possibly adding some smart switching to enable and disable active signals might be something to keep in mind if we see just adding this signal changes some timing at all.

EliotGann commented on rsoxs/Functions/energyscancore.py:
```diff
@@ -1062,7 +1062,7 @@ def flyer_scan_energy(scan_params, sigs=[], md={},locked=True,polarization=0):
 
     """
-    detectors = [Beamstop_WAXS_int, Beamstop_SAXS_int, Izero_Mesh_int, Sample_TEY_int]
+    detectors = [Beamstop_WAXS_int, Beamstop_SAXS_int, DM7_Photodiode_int, Izero_Mesh_int, Sample_TEY_int]
```
here is where we might want to pair down the number of things we actually fly... this is what's causing the huge delay when we end a scan or RE.stop()  I think adding the exact channels we want would be better...

EliotGann replied:
like only one of the beamstops or photodiode is every actually used, and often for transmission we don't want sample_tey, but for TEY we don't want any diodes...

PriyankaKetkarBNL replied:
I didn't change anything here yet other than updating the signal name.  I need to trace the code more to figure out how to identify which beamstop is actually being used and then to set that as the desired detector.  Does it make sense to pull the other changes for now (assuming they all look okay) and then having this as a new pull request later?

EliotGann replied:
for sure, just pointing out if there is a longer delay, this might be why

PriyankaKetkarBNL commented:
I changed the signal name and propagated that throughout.  I also added the large photodiode signal to the fly_find_fiducials and buildeputable functions.  I still have not added it to the spiral scans, as the camera would be in the way.  I plan to look into smart switching for the longer term.

EliotGann commented:
Seems good.  Let's try it.  I'll make a issue for pairing down the signals to just those needed... I think tying it to the configuration, and adding a downstreamNEXAFS configuration is probably the way to go...
